src/sentry/snuba/outcomes.py

```diff
@@ -263,40 +271,58 @@ def __init__(
             query_groupby.update(groupby.get_snuba_groupby())
         self.query_groupby = list(query_groupby)
 
-        self.conditions, self.filter_keys = get_filter(query, params)
+        self.group_by = []
+        for key in self.query_groupby:
+            self.group_by.append(Column(key))
+
+        conditions, filter_keys = get_filter(query, params)
+
+        for org in filter_keys.get("org_id"):
+            self.conditions.append(
+                Condition(Column("org_id"), Op.EQ, org),
+            )
+        if filter_keys.get("project_id") is not None:
+            self.conditions.append(
+                Condition(Column("project_id"), Op.IN, filter_keys.get("project_id")),
+            )
+        for condition in conditions:
+            self.conditions.append(Condition(Column(condition[0]), Op.IN, condition[2]))
+
+        self.select_params = []
+        for aggregation in aggregations:
+            aggregation_field = aggregation[2] if aggregation[1] == "" else aggregation[1]
+            self.select_params.append(
+                Function(aggregation[0], [Column(aggregation_field)], aggregation[2])
+            )
 
 
 def run_outcomes_query_totals(query: QueryDefinition) -> ResultSet:
-    result = raw_query(
-        dataset=query.dataset,
-        start=query.start,
-        end=query.end,
-        groupby=query.query_groupby,
-        aggregations=query.aggregations,
-        rollup=query.rollup,
-        filter_keys=query.filter_keys,
-        conditions=query.conditions,
-        selected_columns=query.query_columns,
-        referrer="outcomes.totals",
-        limit=10000,
+    snql_query = Query(
+        dataset=query.dataset["dataset"].value,
+        match=Entity(query.dataset["match"]),
```

Comment:
probably best to have match be another parameter on the QueryDefinition object -- then you don't have to return a dict in outcomes_dataset, (can return a tuple and set both the members inline)